test/functional/tests/memory/test_check_number_of_processes.py

```diff
@@ -0,0 +1,79 @@
+#
+# Copyright(c) 2020 Intel Corporation
+# SPDX-License-Identifier: BSD-3-Clause-Clear
+#
+
+import pytest
+
+from api.cas import casadm
+from core.test_run import TestRun
+from storage_devices.disk import DiskType, DiskTypeSet, DiskTypeLowerThan
+from test_utils.size import Size, Unit
+
+cleaning_threads_expected = 1
+fill_threads_expected = 0
+metadata_updater_threads_expected = 1
+
+
+@pytest.mark.require_disk("cache", DiskTypeSet([DiskType.optane, DiskType.nand]))
+@pytest.mark.require_disk("core", DiskTypeLowerThan("cache"))
+def test_check_number_of_processes():
+    """
+        title: Check the number of processes created by CAS.
+        description: |
+          For created CAS device check number of IO threads, cleaning threads, fill threads,
+          metadata updater threads.
+        pass_criteria:
+          - Successful CAS creation.
+          - Successful validation of each thread number.
+    """
+    with TestRun.step("Prepare CAS device."):
+        cache_disk = TestRun.disks['cache']
+        cache_disk.create_partitions([Size(2, Unit.GibiByte)])
+        cache_dev = cache_disk.partitions[0]
+
+        core_disk = TestRun.disks['core']
+        core_disk.create_partitions([Size(2, Unit.GibiByte)])
+        core_dev = core_disk.partitions[0]
+
+        cache = casadm.start_cache(cache_dev, force=True)
+        core = cache.add_core(core_dev)
+
+    with TestRun.step("Check number of IO threads."):
+        cmd = f"cat /proc/cpuinfo | grep processor | wc -l"
+        output = TestRun.executor.run(cmd).stdout
+        number_of_processors = int(output)
+        io_threads_actual = get_number_of_processes("cas_io")
+
+        validate_threads_number("IO threads", io_threads_actual, (number_of_processors + 1))
```

Comment:
why +1?

Reply:
That was something from old test and I've checked that those 2 values differ by 1 indeed so for comparison the +1 is needed.

Reply:
additional info: processors from /proc/cpuinfo are zero-based which explains the 1 difference between values.